packages/toolpad-app/src/components/EditableText.tsx

```diff
@@ -73,15 +75,21 @@ const EditableText = React.forwardRef<HTMLInputElement, EditableTextProps>(
 
     const handleChange = React.useCallback(
       (event: React.ChangeEvent<HTMLInputElement>) => {
+        if (readOnly) {
```

Comment:
If this gets called while the input is readonly, that would mean there is an error in the code, right?
In that case, would it make more sense to make this an invariant? 


```suggestion
        invariant(!readOnly, 'Readonly input should be disabled')
```